packages/blade/src/components/Checkbox/CheckboxGroup/useCheckboxGroup.ts

```diff
@@ -0,0 +1,89 @@
+/* eslint-disable @typescript-eslint/no-shadow */
+/* eslint-disable @typescript-eslint/prefer-nullish-coalescing */
+/* eslint-disable @typescript-eslint/explicit-module-boundary-types */
+/* eslint-disable @typescript-eslint/explicit-function-return-type */
+
+import React from 'react';
+import { useControllableState } from '../../../hooks/useControllable';
+import { useId } from '../../../hooks/useId';
+import { useTheme } from '../../BladeProvider';
+import type { CheckboxGroupProps } from './CheckboxGroup';
+
+type UseCheckboxGroupProps = Pick<
+  CheckboxGroupProps,
+  | 'isDisabled'
+  | 'labelPosition'
+  | 'validationState'
+  | 'neccessityIndicator'
+  | 'name'
+  | 'value'
+  | 'defaultValue'
+  | 'onChange'
+>;
+
+const useCheckboxGroup = ({
+  value,
+  defaultValue,
+  isDisabled,
+  labelPosition,
+  onChange,
+  neccessityIndicator,
+  validationState,
+  name,
+}: UseCheckboxGroupProps) => {
+  const { platform } = useTheme();
+  const idBase = useId('checkbox-group');
+  const labelId = `${idBase}-label`;
+  const [checkedValues, setValue] = useControllableState({
+    value,
+    defaultValue: defaultValue || [],
+    onChange,
+  });
+
+  const state = React.useMemo(() => {
+    return {
+      value: checkedValues,
+      setValue(value: string[]) {
+        if (isDisabled) {
+          return;
+        }
+
+        setValue(value);
+      },
+      isChecked(value: string) {
+        return checkedValues.includes(value);
+      },
+      addValue(value: string) {
+        if (isDisabled) {
+          return;
+        }
+        if (!checkedValues.includes(value)) {
+          setValue(checkedValues.concat(value));
+        }
+      },
+      removeValue(value: string) {
+        if (isDisabled) {
+          return;
+        }
+        if (checkedValues.includes(value)) {
+          setValue(checkedValues.filter((existingValue) => existingValue !== value));
+        }
+      },
+    };
+  }, [checkedValues, isDisabled, setValue]);
+
+  const contextValue = React.useMemo(() => {
+    return {
+      validationState,
+      neccessityIndicator,
+      isDisabled,
+      labelPosition: platform === 'onMobile' ? 'top' : labelPosition,
+      name,
+      state,
+    };
+  }, [validationState, neccessityIndicator, isDisabled, platform, labelPosition, name, state]);
+
+  return { state, contextValue, ids: { labelId } };
```

Comment:
why do you call it ids? it's a single id right?

Reply:
Just to have a consistent return type for both useCheckbox and useCheckboxGroup.

Reply:
These "ids" has nothing to do with if it's a group or not. 

These IDs are just accessibility ID references which I need to provide screen reader users proper announcements for helpText and errorText. 

:p check the code. The group has only one id (labelId) and the individual checkbox has two (helpTextId, errorTextId)

Reply:
Yes yes got it. Just to make the return type consistent thats all.